src/include/SIMD.h

```diff
@@ -0,0 +1,3 @@
+#pragma once
+
+enum class SIMD { Scalar, AVX512 };
```

Comment:
Maybe rename `Scalar` to `None` since scalar execution is not SIMD.

Reply:
Done